modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogService.java

```diff
@@ -63,83 +56,49 @@ public interface CatalogService extends EventProducer<CatalogEvent, CatalogEvent
+     * @param timestamp The point-in-time to retrieve the actual catalog version.
+     * @return The active catalog version at the specified timestamp.
      */
-    @Nullable CatalogTableDescriptor table(int tableId, int catalogVersion);
-
-    Collection<CatalogTableDescriptor> tables(int catalogVersion);
+    int activeCatalogVersion(long timestamp);
 
     /**
-     * Returns a descriptor for <em>alive</em> index by the given schema name and index name at given timestamp,
-     * that is an index that has not been dropped yet at a given point in time.
+     * Returns the earliest registered version of the catalog.
      *
-     * <p>This effectively means that the index must be present in the Catalog and not in the {@link CatalogIndexStatus#STOPPING}
-     * state.
+     * @return The earliest registered version of the catalog.
      */
-    @Nullable CatalogIndexDescriptor aliveIndex(String schemaName, String indexName, long timestamp);
-
-    @Nullable CatalogIndexDescriptor index(int indexId, long timestamp);
+    int earliestCatalogVersion();
 
     /**
-     * Returns index descriptor by the given index ID and catalog version.
+     * Returns the latest registered version of the catalog.
      *
-     * @return Index descriptor or {@code null} if index not found.
+     * @return The latest registered version of the catalog.
      */
-    @Nullable CatalogIndexDescriptor index(int indexId, int catalogVersion);
-
-    Collection<CatalogIndexDescriptor> indexes(int catalogVersion);
-
-    List<CatalogIndexDescriptor> indexes(int catalogVersion, int tableId);
-
-    @Nullable CatalogSchemaDescriptor schema(int catalogVersion);
-
-    @Nullable CatalogSchemaDescriptor schema(@Nullable String schemaName, int catalogVersion);
-
-    @Nullable CatalogSchemaDescriptor schema(int schemaId, int catalogVersion);
-
-    @Nullable CatalogZoneDescriptor zone(String zoneName, long timestamp);
-
-    @Nullable CatalogZoneDescriptor zone(int zoneId, long timestamp);
-
-    @Nullable CatalogZoneDescriptor zone(int zoneId, int catalogVersion);
-
-    Collection<CatalogZoneDescriptor> zones(int catalogVersion);
-
-    @Nullable CatalogSchemaDescriptor activeSchema(long timestamp);
-
-    @Nullable CatalogSchemaDescriptor activeSchema(@Nullable String schemaName, long timestamp);
-
-    int activeCatalogVersion(long timestamp);
-
-    /** Returns the earliest registered version of the catalog. */
-    int earliestCatalogVersion();
-
-    /** Returns the latest registered version of the catalog. */
     int latestCatalogVersion();
 
     /**
      * Returns a future, which completes, when catalog of given version will be available.
      *
-     * @param version Catalog version to wait for.
+     * @param version The catalog version to wait for.
+     * @return A future that completes when the catalog of the given version becomes available.
      */
     CompletableFuture<Void> catalogReadyFuture(int version);
 
```

Comment:
perhaps, it worth to move `catalogInitializationFuture` method to `CatalogManager` interface or even implementation class. WDYT?